File: pathfinder-core/src/main/java/de/cubbossa/pathfinder/PathFinderExtensionBase.java
package de.cubbossa.pathfinder;

import de.cubbossa.pathapi.PathFinderExtension;
import lombok.Getter;
import lombok.Setter;

@Setter
@Getter
public abstract class PathFinderExtensionBase implements PathFinderExtension {

  private boolean disabled = false;

  @Override
  public void disable() {
    disabled = true;
  }
}

File: pathfinder-core/src/main/java/de/cubbossa/pathfinder/PathFinderExtensionBase.kt
package de.cubbossa.pathfinder

import lombok.Getter
import lombok.Setter

@Setter
@Getter
abstract class PathFinderExtensionBase : PathFinderExtension {
    private var disabled = false

    override fun disable() {
        disabled = true
    }

    override fun isDisabled(): Boolean {
        return disabled
    }
}
